services/transactionService.js

```diff
@@ -509,18 +530,29 @@ export const generatepdf = async (req, res) => {
       bufferToPDF(pdfBuffer, outputPath);
     });
 
+
     async function bufferToPDF(pdfBuffer, pdfPath) {
       try {
         fs.writeFileSync(pdfPath, pdfBuffer);
         console.log('PDF file has been written successfully');
+        const invoiceData = {
+          InvoiceNo: invoiceNumber,
+          InvoiceDate: formattedDate,
+          CustomerID: customerId,
+          TotalVehicles: totalVehicles,
+          TotalAmount: total,
+          // ReceivedAmount: 800.00, // Partial payment received
+          // Discount: 50 // Discount given
+        };
 
-
+        const newInvoice = await Invoice.create(invoiceData);
+        console.log('Invoice created successfully:', newInvoice);
 
       } catch (err) {
         console.error('Error writing PDF to file:', err);
       }
```

Comment:
If this function fails the API will still return 200 which is not correct. Either remove this try-catch block or return a 500 response in the catch block.